driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -73,20 +76,33 @@ public void apply(final ServerSettings.Builder builder) {
                 .build();
 
         clientUnderTest = MongoClients.create(settings);
-
         failPointClient = MongoClients.create(getMongoClientSettingsBuilder().build());
         stepDownClient = MongoClients.create(getMongoClientSettingsBuilder().build());
 
+        try {
+            originalPrimary = Fixture.getPrimary();
+        } catch (InterruptedException e) {
+        }
         clientDatabase = clientUnderTest.getDatabase(DATABASE_NAME);
         clientCollection = clientDatabase.getCollection(COLLECTION_NAME);
     }
 
     @After
     public void cleanUp() {
         if (failPointClient != null) {
-            MongoDatabase failPointAdminDB = failPointClient.getDatabase("admin");
-            failPointAdminDB.runCommand(
+            failPointClient.close();
```

Comment:
Close this MongoClient after joining the stepdownThread, which is using it.

Reply:
The stepDownThread is using the stepDownClient, not the failPointClient.